streaming/src/main/java/org/openmrs/analytics/FhirEventWorker.java

```diff
@@ -34,52 +29,49 @@
 	
 	private static final Logger log = LoggerFactory.getLogger(FhirEventWorker.class);
 	
-	private String feedBaseUrl;
-	
-	private String jSessionId;
+	private String sourceUrl;
 	
 	private String resourceType;
 	
 	private Class<T> resourceClass;
 	
-	private FhirContext fhirContext = FhirContext.forR4();
-	
 	private FhirStoreUtil fhirStoreUtil;
 	
-	FhirEventWorker(String feedBaseUrl, String jSessionId, String resourceType, Class<T> resourceClass,
-	    String gcpFhirStore) {
-		this.feedBaseUrl = feedBaseUrl;
-		this.jSessionId = jSessionId;
-		this.resourceType = resourceType;
+	private String sourceUser;
+	
+	private String sourcePw;
+	
+	FhirEventWorker(String sourceUrl, Class<T> resourceClass, String sourceUser, String sourcePW, String targetUrl) {
+		this.sourceUrl = sourceUrl;
 		this.resourceClass = resourceClass;
-		// TODO inject dependency.
-		this.fhirStoreUtil = new FhirStoreUtil(gcpFhirStore);
+		this.sourceUser = sourceUser;
+		this.sourcePw = sourcePW;
+		
+		this.fhirStoreUtil = new FhirStoreUtil(targetUrl, sourceUrl, sourceUser, sourcePW);
 	}
 	
-	private String fetchFhirResource(String urlStr) {
+	private T fetchFhirResource(String resourceUrl) {
```

Comment:
Given that now you have switched to using HAPI FHIR clients (which is great) I think we can try write unit-tests for this class and in particular this method (by using a mock client). Can you please either add them in this PR (preferred) or add a TODO for doing so in future?